writer.go

```diff
@@ -262,6 +265,8 @@ func (w *Writer) WriteMessages(ctx context.Context, msgs ...Message) error {
 		}
 
 		for _, msg := range msgs {
+			msg.CompressionCodec = w.config.CompressionCodec
```

Comment:
Should we check for `msg.CompressionCodec == nil` here? That way we allow individual messages to overwrite the compression codec they use, and otherwise default to the one configured on the writer.

Reply:
Does it mean we are going to allow multiple codec in the same topic ? Feels weird even so Kafka should be able to handle that.